### PR TITLE
Refactor ProductDataProvider to use atomFamily and default jotai store instead of per-product stores

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

Use jotai atomFamily to store current product data instead of creating new store to reset data for new context

## Justify why they are needed

Next step would be unifying Jotai atoms to single providers, this would be needed for app router

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3929** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3936** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936?utm_source=stack-comment).

**Review by gustaveen — APPROVED** [review]
Looks good to me! But @guilhermespopolin you probably have better insight on the issues we had with this before

File: apps/store/src/components/ProductData/ProductDataProvider.tsx [hunk 1]
```diff
@@ -1,14 +1,28 @@
...
 
-const SELECTED_PRODUCT_VARIANT_ATOM = atom((get) => {
-  const productData = get(PRODUCT_DATA_ATOM)
-  const typeOfContract = get(SELECTED_TYPE_OF_CONTRACT_ATOM)
+const currentProductAtom = atom<string | null>(null)
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const productDataAtomFamily = atomFamily((_key) => atom<ProductData | null>(null))
+const productDataAtom = atom<ProductData | null>((get) => {
+  const key = get(currentProductAtom)
+  if (key == null) return null
```
Comment (guilhermespopolin) [inline apps/store/src/components/ProductData/ProductDataProvider.tsx:17]: nit:
Based on `get` return type, we can use `===`. Or are you being extra safe? 🤔

**Review by guilhermespopolin — APPROVED** [review]
Nice improvement :)

**Comment by alebedev** [comment]
### Merge activity

* **Feb 20, 5:01 AM EST**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3936).

**Pull request merged**